src/index.ts

```diff
@@ -90,11 +90,18 @@ export async function findManyCursorConnection<
   const endCursor =
     records.length > 0 ? encodeCursor(records[records.length - 1], options) : undefined
 
+  // Allow the recordToEdge function to return a custom edge type which will be inferred
+  type EdgeExtended = typeof options.recordToEdge extends (record: Record) => infer X
+    ? X extends CustomEdge
+      ? X & { cursor: string }
+      : CustomEdge
+    : CustomEdge
```

Comment:
Rough gist is that we check if theres a recordToEdge fn, if it exists then take the return value and call it `X` - if X is a type which CustomEdge can entirely fit inside then return that! Otherwise return the CustomEdge

Should be fully backwards compatible, as evidenced by the rest of the codebase compiling and the new tests below